packages/contracts/contracts/plugin/examples/count/CounterV1PluginManager.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier:    MIT
+
+pragma solidity 0.8.10;
+
+import "../../PluginManager.sol";
+import "./MultiplyHelper.sol";
+import "./CounterV1.sol";
+
+contract CounterV1PluginManager is PluginManager {
+    
+    MultiplyHelper private multiplyHelperBase;
+    CounterV1 private counterBase;
+
+    address private constant NO_ORACLE = address(0);
+
+    constructor() {
+        multiplyHelperBase = new MultiplyHelper();
+        counterBase = new CounterV1();
+    }
+
+    function deploy(address dao, bytes memory data)
+        external
+        virtual
+        override
+        returns (address plugin, address[] memory relatedContracts)
+    {
+        (address _multiplyHelper, uint256 _num) = abi.decode(data, (address, uint256));
+
+        relatedContracts = new address[](1);
+
+        if (_multiplyHelper == address(0)) {
+            _multiplyHelper = createProxy(dao, address(multiplyHelperBase), "0x");
+        }
+
+        bytes memory init = abi.encodeWithSelector(
+            bytes4(keccak256("function initialize(address,uint256))")),
+            _multiplyHelper,
+            _num
+        );
+
+        relatedContracts[0] = _multiplyHelper;
+
+        plugin = createProxy(dao, getImplementationAddress(), init);
```

Comment:
```suggestion
        // Decode the parameters from the UI
        (address _multiplyHelper, uint256 _num) = abi.decode(data, (address, uint256));

        // Allocate space for one address for the helper contracts for which we
        // also want to define permissions
        relatedContracts = new address[](1);

        if (_multiplyHelper == address(0)) {
            // Deploy some internal helper contract for the Plugin
            _multiplyHelper = createProxy(dao, address(multiplyHelperBase), "0x");
        }

        // Encode the parameters that will be passed to initialize() on the Plugin
        bytes memory initData = abi.encodeWithSelector(
            bytes4(keccak256("function initialize(address,uint256))")),
            _multiplyHelper,
            _num
        );

        // Address of the helper so that PluginInstaller can grant the requested permissions on it
        relatedContracts[0] = _multiplyHelper;

        // Deploy the Plugin itself, make it point to the implementation and
        // pass it the initialization params
        plugin = createProxy(dao, getImplementationAddress(), initData);
```